src/transformations/utils/build-super-calls-for-chain.ts

```diff
@@ -110,6 +109,32 @@ function isImplicitlyConstructed(contract: ContractDefinition): boolean {
 
   return (
     contract.contractKind === 'contract' &&
-    (ctor == undefined || ctor.parameters.parameters.length === 0)
+    (ctor == undefined ||
+      ctor.parameters.parameters.length === 0 )
+  );
+}
+
+function isEmpty(contract: ContractDefinition): boolean {
```

Comment:
Why not add
`if (contract.contractKind !== 'contract) { return false; }`
at the begining?

That would somehow simplify the return at the end, and give more confidence about the object we are working on.

Reply:
I think this all function would be clearer if it was written

```
function isEmpty(contract: ContractDefinition): boolean {
  if (contract.contractKind !== 'contract') { return false; }

  const ctor = getConstructor(contract);

  const hasVarInitNodes = [...findAll('VariableDeclaration', contract)].some(
    v => v.stateVariable && v.value && !v.constant && !hasOverride(v, 'state-variable-assignment'),
  );
  const hasModifiers = ctor.modifiers.some(
    call => call.modifierName.referencedDeclaration != null && !ctor.linearizedBaseContracts.includes(call.modifierName.referencedDeclaration)
  );

  return ctor?.body?.statements?.length == undefined && !hasVarInitNodes && !hasModifiers;
}
```